File: 算法训练营/20-字符串算法/771/迭代法.go
/*
 * @lc app=leetcode.cn id=771 lang=golang
 *
 * [771] 宝石与石头
 */

// @lc code=start
func numJewelsInStones(J string, S string) int {
	var arr [256]rune
	for _, ch := range J {
		arr[ch] = 1
	}

	count := 0
	for _, ch := range S {
		if arr[ch] > 0 {
			count++
		}
	}
	return count
}
// @lc code=end

File: 算法训练营/20-字符串算法/771/迭代法.py

#
# @lc app=leetcode.cn id=771 lang=python3
#
# [771] 宝石与石头
#

# @lc code=start


class Solution:
    def numJewelsInStones(self, J: str, S: str) -> int:
        count = 0
        for s in S:
            if s in J:
                count += 1
        return count

# @lc code=end
